vsphere_virtual_machine crash

Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.
### Terraform Version

`Terraform v0.7.0`
### Affected Resource(s)

Please list the resources as a list, for example:
- vsphere_virtual_machine

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.
### Terraform Configuration Files

```
provider "vsphere" {
  user           = "${var.domain_user}"
  password       = "${var.domain_pass}"
  vsphere_server = "${lookup(var.vcenter_server,var.domain)}"
  allow_unverified_ssl = true
}

resource "vsphere_virtual_machine" "server" {
  count = "${var.instances}"
  name   = "${element(split(",", var.hostnames), count.index)}"
  vcpu   = "${var.cpus}"
  memory = "${var.memory}"
  time_zone = "UTC"
  datacenter = "${lookup(var.vcenter_datacenter,var.domain)}"
  cluster = "${lookup(var.cluster,var.domain)}"
  domain = "${lookup(var.dns_domain,var.domain)}"
  dns_servers = ["${split(",", lookup(var.dns_servers,var.domain))}"]
  dns_suffixes = ["${split(",", lookup(var.dns_suffixes,var.domain))}"]
  gateway = "${lookup(var.gateway,var.domain)}"

  network_interface {
    label = "${lookup(var.network,var.domain)}"
    ipv4_address = "${element(split(",", var.ip_addresses), count.index)}"
    ipv4_prefix_length = "${lookup(var.prefix_length,var.domain)}"
  }

  disk {
    template = "${lookup(var.template,var.domain)}"
    type = "thin"
    datastore = "${lookup(var.datastore,var.domain)}"
  }

  custom_configuration_parameters {
    role = "${var.role}"
  }
}
```
### Debug Output

The entire crash.log can be found here: https://gist.github.com/kzittritsch/e6e7e7b4f0cbf6ad279633f428267826
### Panic Output

```
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] Datacenter - &object.Datacenter{Common:object.Common{c:(*vim25.Client)(0xc820442500), r:types.ManagedObjectReference{Type:"Datacenter", Value:"datacenter-21"}}}
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] mvm.Summary.Config - types.VirtualMachineConfigSummary{DynamicData:types.DynamicData{}, Name:"la1ad-rtbstats01", Template:false, VmPathName:"[vol_vmstor_05_SAS_01] la1ad-rtbstats01/la1ad-rtbstats01.vmx", MemorySizeMB:4096, CpuReservation:0, MemoryReservation:0, NumCpu:2, NumEthernetCards:1, NumVirtualDisks:1, Uuid:"420e6df7-f2b7-a649-7567-4705f706b7e6", InstanceUuid:"500e43ba-7ad0-d258-ce90-62dd24dbd7d6", GuestId:"rhel7_64Guest", GuestFullName:"Red Hat Enterprise Linux 7 (64-bit)", Annotation:"", Product:(*types.VAppProductInfo)(0xc824f96fa0), InstallBootRequired:(*bool)(0xc82554c65d), FtInfo:types.BaseFaultToleranceConfigInfo(nil), ManagedBy:(*types.ManagedByInfo)(nil)}
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] mvm.Summary.Config - &types.VirtualMachineConfigInfo{DynamicData:types.DynamicData{}, ChangeVersion:"2016-08-06T15:49:15.708957Z", Modified:time.Time{sec:62135596800, nsec:0, loc:(*time.Location)(0x571a6c0)}, Name:"la1ad-rtbstats01", GuestFullName:"Red Hat Enterprise Linux 7 (64-bit)", Version:"vmx-09", Uuid:"420e6df7-f2b7-a649-7567-4705f706b7e6", InstanceUuid:"500e43ba-7ad0-d258-ce90-62dd24dbd7d6", NpivNodeWorldWideName:[]int64(nil), NpivPortWorldWideName:[]int64(nil), NpivWorldWideNameType:"", NpivDesiredNodeWwns:0, NpivDesiredPortWwns:0, NpivTemporaryDisabled:(*bool)(0xc825783d58), NpivOnNonRdmDisks:(*bool)(nil), LocationId:"564de364-5e0b-1659-1f5c-4de16fc93c1c", Template:false, GuestId:"rhel7_64Guest", AlternateGuestName:"", Annotation:"", Files:types.VirtualMachineFileInfo{DynamicData:types.DynamicData{}, VmPathName:"[vol_vmstor_05_SAS_01] la1ad-rtbstats01/la1ad-rtbstats01.vmx", SnapshotDirectory:"[vol_vmstor_05_SAS_01] la1ad-rtbstats01/", SuspendDirectory:"[vol_vmstor_05_SAS_01] la1ad-rtbstats01/", LogDirectory:"[vol_vmstor_05_SAS_01] la1ad-rtbstats01/", FtMetadataDirectory:""}, Tools:(*types.ToolsConfigInfo)(0xc8250312c0), Flags:types.VirtualMachineFlagInfo{DynamicData:types.DynamicData{}, DisableAcceleration:(*bool)(0xc825694025), EnableLogging:(*bool)(0xc825694035), UseToe:(*bool)(0xc82569406a), RunWithDebugInfo:(*bool)(0xc825694095), MonitorType:"release", HtSharing:"any", SnapshotDisabled:(*bool)(0xc82569411b), SnapshotLocked:(*bool)(0xc82569414d), DiskUuidEnabled:(*bool)(0xc82569418d), VirtualMmuUsage:"automatic", VirtualExecUsage:"hvAuto", SnapshotPowerOffBehavior:"powerOff", RecordReplayEnabled:(*bool)(0xc825694270), FaultToleranceType:""}, ConsolePreferences:(*types.VirtualMachineConsolePreferences)(nil), DefaultPowerOps:types.VirtualMachineDefaultPowerOpInfo{DynamicData:types.DynamicData{}, PowerOffType:"soft", SuspendType:"soft", ResetType:"soft", DefaultPowerOffType:"soft", DefaultSuspendType:"hard", DefaultResetType:"soft", StandbyAction:"checkpoint"}, Hardware:types.VirtualHardware{DynamicData:types.DynamicData{}, NumCPU:2, NumCoresPerSocket:1, MemoryMB:4096, VirtualICH7MPresent:(*bool)(0xc82569445c), VirtualSMCPresent:(*bool)(0xc82569446d), Device:[]types.BaseVirtualDevice{(*types.VirtualIDEController)(0xc824fdd2d0), (*types.VirtualIDEController)(0xc824fdd340), (*types.VirtualPS2Controller)(0xc824fdd3b0), (*types.VirtualPCIController)(0xc824fdd420), (*types.VirtualSIOController)(0xc824fdd490), (*types.VirtualKeyboard)(0xc825210550), (*types.VirtualPointingDevice)(0xc8252105f0), (*types.VirtualMachineVideoCard)(0xc824b817a0), (*types.VirtualMachineVMCIDevice)(0xc824fdd500), (*types.VirtualLsiLogicController)(0xc824b818c0), (*types.VirtualCdrom)(0xc825210690), (*types.VirtualDisk)(0xc82522bf40), (*types.VirtualVmxnet3)(0xc824f96000)}}, CpuAllocation:(*types.ResourceAllocationInfo)(0xc8246d7650), MemoryAllocation:(*types.ResourceAllocationInfo)(0xc8246d7710), LatencySensitivity:(*types.LatencySensitivity)(0xc82585b2c0), MemoryHotAddEnabled:(*bool)(0xc825695fa6), CpuHotAddEnabled:(*bool)(0xc825695fb4), CpuHotRemoveEnabled:(*bool)(0xc825695ff4), HotPlugMemoryLimit:65536, HotPlugMemoryIncrementSize:128, CpuAffinity:(*types.VirtualMachineAffinityInfo)(nil), MemoryAffinity:(*types.VirtualMachineAffinityInfo)(nil), NetworkShaper:(*types.VirtualMachineNetworkShaperInfo)(nil), ExtraConfig:[]types.BaseOptionValue{(*types.OptionValue)(0xc82585b6e0), (*types.OptionValue)(0xc82585b880), (*types.OptionValue)(0xc82585b9e0), (*types.OptionValue)(0xc82585bb40), (*types.OptionValue)(0xc82585bd00), (*types.OptionValue)(0xc82585be60), (*types.OptionValue)(0xc82585bfa0), (*types.OptionValue)(0xc82584e100), (*types.OptionValue)(0xc82584e2c0), (*types.OptionValue)(0xc82584e420), (*types.OptionValue)(0xc82584e5a0), (*types.OptionValue)(0xc82584e720), (*types.OptionValue)(0xc82584e880), (*types.OptionValue)(0xc82584ea20), (*types.OptionValue)(0xc82584eb80), (*types.OptionValue)(0xc82584ece0), (*types.OptionValue)(0xc82584eea0), (*types.OptionValue)(0xc82584f000), (*types.OptionValue)(0xc82584f180), (*types.OptionValue)(0xc82584f2c0), (*types.OptionValue)(0xc82584f420), (*types.OptionValue)(0xc82584f5c0), (*types.OptionValue)(0xc82584f6e0), (*types.OptionValue)(0xc82584f800), (*types.OptionValue)(0xc82584f940), (*types.OptionValue)(0xc82584fa80), (*types.OptionValue)(0xc82584fbe0), (*types.OptionValue)(0xc82584fd20), (*types.OptionValue)(0xc82584fe40), (*types.OptionValue)(0xc82584ff40), (*types.OptionValue)(0xc82583e0a0), (*types.OptionValue)(0xc82583e200), (*types.OptionValue)(0xc82583e3c0), (*types.OptionValue)(0xc82583e520), (*types.OptionValue)(0xc82583e6a0), (*types.OptionValue)(0xc82583e820), (*types.OptionValue)(0xc82583e980), (*types.OptionValue)(0xc82583eb20), (*types.OptionValue)(0xc82583ec80), (*types.OptionValue)(0xc82583eda0), (*types.OptionValue)(0xc82583ef00), (*types.OptionValue)(0xc82583f060), (*types.OptionValue)(0xc82583f1c0), (*types.OptionValue)(0xc82583f360), (*types.OptionValue)(0xc82583f4c0), (*types.OptionValue)(0xc82583f640)}, CpuFeatureMask:[]types.HostCpuIdInfo(nil), DatastoreUrl:[]types.VirtualMachineConfigInfoDatastoreUrlPair{types.VirtualMachineConfigInfoDatastoreUrlPair{DynamicData:types.DynamicData{}, Name:"vol_vmstor_05_SAS_01", Url:"/vmfs/volumes/b81fad2e-b8d37232"}}, SwapPlacement:"inherit", BootOptions:(*types.VirtualMachineBootOptions)(0xc825210960), FtInfo:types.BaseFaultToleranceConfigInfo(nil), RepConfig:(*types.ReplicationConfigSpec)(nil), VAppConfig:(*types.VmConfigInfo)(0xc8252f6680), VAssertsEnabled:(*bool)(0xc825619cc9), ChangeTrackingEnabled:(*bool)(0xc825619cfd), Firmware:"bios", MaxMksConnections:40, GuestAutoLockEnabled:(*bool)(0xc825619d5a), ManagedBy:(*types.ManagedByInfo)(nil), MemoryReservationLockedToMax:(*bool)(0xc825619d6d), InitialOverhead:(*types.VirtualMachineConfigInfoOverheadInfo)(0xc825619d90), NestedHVEnabled:(*bool)(0xc825619dc9), VPMCEnabled:(*bool)(0xc825619e4b), ScheduledHardwareUpgradeInfo:(*types.ScheduledHardwareUpgradeInfo)(0xc8253d7540), ForkConfigInfo:(*types.VirtualMachineForkConfigInfo)(nil), VFlashCacheReservation:0, VmxConfigChecksum:[]uint8(nil), MessageBusTunnelEnabled:(*bool)(nil), VmStorageObjectId:"", SwapStorageObjectId:""}
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] mvm.Guest.Net - []types.GuestNicInfo{types.GuestNicInfo{DynamicData:types.DynamicData{}, Network:"", IpAddress:[]string(nil), MacAddress:"26:ab:9b:f1:2f:59", Connected:true, DeviceConfigId:-1, DnsConfig:(*types.NetDnsConfigInfo)(nil), IpConfig:(*types.NetIpConfigInfo)(0xc8246d78c0), NetBIOSConfig:types.BaseNetBIOSConfigInfo(nil)}, types.GuestNicInfo{DynamicData:types.DynamicData{}, Network:"Server", IpAddress:[]string{"10.14.11.24"}, MacAddress:"00:50:56:8e:1a:82", Connected:true, DeviceConfigId:4000, DnsConfig:(*types.NetDnsConfigInfo)(nil), IpConfig:(*types.NetIpConfigInfo)(0xc8246d7950), NetBIOSConfig:types.BaseNetBIOSConfigInfo(nil)}, types.GuestNicInfo{DynamicData:types.DynamicData{}, Network:"", IpAddress:[]string{"172.17.0.1"}, MacAddress:"02:42:83:d3:08:5b", Connected:true, DeviceConfigId:-1, DnsConfig:(*types.NetDnsConfigInfo)(nil), IpConfig:(*types.NetIpConfigInfo)(0xc8246d79b0), NetBIOSConfig:types.BaseNetBIOSConfigInfo(nil)}}
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] resourceVSphereVirtualMachineRead - Analyzing disk: [vol_vmstor_05_SAS_01] la1ad-rtbstats01/la1ad-rtbstats01.vmdk
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] disks: []map[string]interface {}{map[string]interface {}{"keep_on_remove":false, "controller_type":"scsi", "template":"centos72-plat-0425-sas01", "type":"thin", "size":0, "name":"", "iops":0, "bootable":false, "datastore":"vol_vmstor_05_SAS_01", "vmdk":"", "uuid":"", "key":0}}
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] v.Network - "Server"
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] p.String - "10.14.11.24"
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] ip.PrefixLength - 20
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] networkInterface: map[string]interface {}{"label":"Server", "mac_address":"00:50:56:8e:1a:82", "ipv4_address":"10.14.11.24", "ipv4_prefix_length":20}
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] networkInterface: map[string]interface {}{"label":"Server", "mac_address":"00:50:56:8e:1a:82", "ipv4_address":"10.14.11.24", "ipv4_prefix_length":20}
2016/08/17 07:11:49 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/17 07:11:49 [DEBUG] ipv4_gateway of device id 1: 10.14.8.1
2016/08/17 07:11:49 [DEBUG] plugin: terraform: panic: runtime error: index out of range
2016/08/17 07:11:49 [DEBUG] plugin: terraform: 
2016/08/17 07:11:49 [DEBUG] plugin: terraform: goroutine 29 [running]:
2016/08/17 07:11:49 [DEBUG] plugin: terraform: panic(0x2fb1ba0, 0xc820010060)
2016/08/17 07:11:49 [DEBUG] plugin: terraform:  /opt/go/src/runtime/panic.go:481 +0x3e6
2016/08/17 07:11:49 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/vsphere.resourceVSphereVirtualMachineRead(0xc820451200, 0x37041e0, 0xc820448fe0, 0x0, 0x0)
2016/08/17 07:11:49 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/vsphere/resource_vsphere_virtual_machine.go:1049 +0x4368
2016/08/17 07:11:49 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc8203b09c0, 0xc820459900, 0x37041e0, 0xc820448fe0, 0xc8203bee70, 0x0, 0x0)
2016/08/17 07:11:49 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:259 +0x432
2016/08/17 07:11:49 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc8203c2e10, 0xc8204598c0, 0xc820459900, 0x1, 0x0, 0x0)
2016/08/17 07:11:49 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:201 +0x1da
2016/08/17 07:11:49 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc8203b4be0, 0xc820449330, 0xc820449d50, 0x0, 0x0)
2016/08/17 07:11:49 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/plugin/resource_provider.go:482 +0x6a
2016/08/17 07:11:49 [DEBUG] plugin: terraform: reflect.Value.call(0x2c6c740, 0x3650738, 0x13, 0x37c3bf8, 0x4, 0xc825a07eb8, 0x3, 0x3, 0x0, 0x0, ...)
2016/08/17 07:11:49 [DEBUG] plugin: terraform:  /opt/go/src/reflect/value.go:435 +0x120d
2016/08/17 07:11:49 [DEBUG] plugin: terraform: reflect.Value.Call(0x2c6c740, 0x3650738, 0x13, 0xc825a07eb8, 0x3, 0x3, 0x0, 0x0, 0x0)
2016/08/17 07:11:49 [DEBUG] plugin: terraform:  /opt/go/src/reflect/value.go:303 +0xb1
2016/08/17 07:11:49 [DEBUG] plugin: terraform: net/rpc.(*service).call(0xc8203af500, 0xc8203af4c0, 0xc8203b2e00, 0xc8203d0c00, 0xc8203b5520, 0x2540fc0, 0xc820449330, 0x16, 0x2541020, 0xc820449d50, ...)
2016/08/17 07:11:49 [DEBUG] plugin: terraform:  /opt/go/src/net/rpc/server.go:383 +0x1c2
2016/08/17 07:11:49 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2016/08/17 07:11:49 [DEBUG] plugin: terraform:  /opt/go/src/net/rpc/server.go:477 +0x49d
```
### Expected Behavior

Correctly resolve gateway for network interface(s)
### Actual Behavior

Crashed when attempting to set the gateway for a seemingly non-existent ("index out of range") network interface
### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
### Important Factoids

We use a couple in-house providers but I don't believe these would cause the issue above.
### References

N/A

Hi @kzittritsch,

I think that we're seeing the same problem, it seems to be caused by an off-by-one error as the change in #8276 fixes it (for me anyway!). 

Pre-patch:

```
2016/08/18 00:37:21 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:37:21 [INFO] Created virtual machine: templates/my-new-vm.dc.domain.com
2016/08/18 00:37:21 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:37:21 [DEBUG] virtual machine resource data: &schema.ResourceData{schema:map[string]*schema.Schema{"datacenter":(*schema.Schema)(0xc8203f2c40), "linked_clone":(*schema.Schema)(0xc8203f2ee0), "vcpu":(*schema.Schema)(0xc8203f29a0), "skip_customization":(*schema.Schema)(0xc8203f3420), "windows_opt_config":(*schema.Schema)(0xc8203f37a0), "network_interface":(*schema.Schema)(0xc8203f3880), "resource_pool":(*schema.Schema)(0xc8203f2e00), "memory_reservation":(*schema.Schema)(0xc8203f2b60), "cluster":(*schema.Schema)(0xc8203f2d20), "gateway":(*schema.Schema)(0xc8203f2fc0), "domain":(*schema.Schema)(0xc8203f30a0), "time_zone":(*schema.Schema)(0xc8203f3180), "dns_suffixes":(*schema.Schema)(0xc8203f3260), "cdrom":(*schema.Schema)(0xc8203f3a40), "memory":(*schema.Schema)(0xc8203f2a80), "folder":(*schema.Schema)(0xc8203f28c0), "dns_servers":(*schema.Schema)(0xc8203f3340), "enable_disk_uuid":(*schema.Schema)(0xc8203f3500), "uuid":(*schema.Schema)(0xc8203f35e0), "custom_configuration_parameters":(*schema.Schema)(0xc8203f36c0), "disk":(*schema.Schema)(0xc8203f3960), "name":(*schema.Schema)(0xc8203f27e0)}, config:(*terraform.ResourceConfig)(nil), state:(*terraform.InstanceState)(nil), diff:(*terraform.InstanceDiff)(0xc8201abbc0), meta:map[string]string(nil), multiReader:(*schema.MultiLevelFieldReader)(0xc8204a8380), setWriter:(*schema.MapFieldWriter)(0xc8204a82e0), newState:(*terraform.InstanceState)(0xc8204aa500), partial:false, partialMap:map[string]struct {}(nil), once:sync.Once{m:sync.Mutex{state:0, sema:0x0}, done:0x1}, isNew:true}
...
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] Datacenter - &object.Datacenter{Common:object.Common{c:(*vim25.Client)(0xc8201daa00), r:types.ManagedObjectReference{Type:"Datacenter", Value:"datacenter-2"}}}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] mvm.Summary.Config - types.VirtualMachineConfigSummary{DynamicData:types.DynamicData{}, Name:"my-new-vm.dc.domain.com", Template:false, VmPathName:"[VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com.vmx", MemorySizeMB:2048, CpuReservation:0, MemoryReservation:0, NumCpu:2, NumEthernetCards:1, NumVirtualDisks:3, Uuid:"42298c43-4916-d536-4b40-c50e25fdb0d5", InstanceUuid:"5029edf7-b993-8289-156f-68d0fd75f481", GuestId:"centos64Guest", GuestFullName:"CentOS 4/5/6/7 (64-bit)", Annotation:"", Product:(*types.VAppProductInfo)(0xc8204e1400), InstallBootRequired:(*bool)(0xc823b3df4d), FtInfo:types.BaseFaultToleranceConfigInfo(nil), ManagedBy:(*types.ManagedByInfo)(nil)}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] Datacenter - &object.Datacenter{Common:object.Common{c:(*vim25.Client)(0xc8201daa00), r:types.ManagedObjectReference{Type:"Datacenter", Value:"datacenter-2"}}}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] mvm.Summary.Config - types.VirtualMachineConfigSummary{DynamicData:types.DynamicData{}, Name:"my-new-vm.dc.domain.com", Template:false, VmPathName:"[VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com.vmx", MemorySizeMB:2048, CpuReservation:0, MemoryReservation:0, NumCpu:2, NumEthernetCards:1, NumVirtualDisks:3, Uuid:"42298c43-4916-d536-4b40-c50e25fdb0d5", InstanceUuid:"5029edf7-b993-8289-156f-68d0fd75f481", GuestId:"centos64Guest", GuestFullName:"CentOS 4/5/6/7 (64-bit)", Annotation:"", Product:(*types.VAppProductInfo)(0xc8204e1400), InstallBootRequired:(*bool)(0xc823b3df4d), FtInfo:types.BaseFaultToleranceConfigInfo(nil), ManagedBy:(*types.ManagedByInfo)(nil)}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] mvm.Summary.Config - &types.VirtualMachineConfigInfo{DynamicData:types.DynamicData{}, ChangeVersion:"2016-08-17T23:37:46.071617Z", Modified:time.Time{sec:62135596800, nsec:0, loc:(*time.Location)(0x571a6c0)}, Name:"my-new-vm.dc.domain.com", GuestFullName:"CentOS 4/5/6/7 (64-bit)", Version:"vmx-10", Uuid:"42298c43-4916-d536-4b40-c50e25fdb0d5", InstanceUuid:"5029edf7-b993-8289-156f-68d0fd75f481", NpivNodeWorldWideName:[]int64(nil), NpivPortWorldWideName:[]int64(nil), NpivWorldWideNameType:"", NpivDesiredNodeWwns:0, NpivDesiredPortWwns:0, NpivTemporaryDisabled:(*bool)(0xc823b381a8), NpivOnNonRdmDisks:(*bool)(nil), LocationId:"564d16de-7f55-37bd-8269-a05dc205eeff", Template:false, GuestId:"centos64Guest", AlternateGuestName:"", Annotation:"", Files:types.VirtualMachineFileInfo{DynamicData:types.DynamicData{}, VmPathName:"[VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com.vmx", SnapshotDirectory:"[VMFS_TGT] my-new-vm.dc.domain.com_1/", SuspendDirectory:"[VMFS_TGT] my-new-vm.dc.domain.com_1/", LogDirectory:"[VMFS_TGT] my-new-vm.dc.domain.com_1/", FtMetadataDirectory:""}, Tools:(*types.ToolsConfigInfo)(0xc8202c87e0), Flags:types.VirtualMachineFlagInfo{DynamicData:types.DynamicData{}, DisableAcceleration:(*bool)(0xc823b38415), EnableLogging:(*bool)(0xc823b38425), UseToe:(*bool)(0xc823b3845a), RunWithDebugInfo:(*bool)(0xc823b38475), MonitorType:"release", HtSharing:"any", SnapshotDisabled:(*bool)(0xc823b384eb), SnapshotLocked:(*bool)(0xc823b3851d), DiskUuidEnabled:(*bool)(0xc823b3854d), VirtualMmuUsage:"automatic", VirtualExecUsage:"hvAuto", SnapshotPowerOffBehavior:"powerOff", RecordReplayEnabled:(*bool)(0xc823b38610), FaultToleranceType:""}, ConsolePreferences:(*types.VirtualMachineConsolePreferences)(nil), DefaultPowerOps:types.VirtualMachineDefaultPowerOpInfo{DynamicData:types.DynamicData{}, PowerOffType:"soft", SuspendType:"soft", ResetType:"soft", DefaultPowerOffType:"soft", DefaultSuspendType:"hard", DefaultResetType:"soft", StandbyAction:"checkpoint"}, Hardware:types.VirtualHardware{DynamicData:types.DynamicData{}, NumCPU:2, NumCoresPerSocket:1, MemoryMB:2048, VirtualICH7MPresent:(*bool)(0xc823b387bc), VirtualSMCPresent:(*bool)(0xc823b387cd), Device:[]types.BaseVirtualDevice{(*types.VirtualIDEController)(0xc8201d75e0), (*types.VirtualIDEController)(0xc8201d7650), (*types.VirtualPS2Controller)(0xc8201d76c0), (*types.VirtualPCIController)(0xc8201d77a0), (*types.VirtualSIOController)(0xc8201d7810), (*types.VirtualKeyboard)(0xc821aa3130), (*types.VirtualPointingDevice)(0xc821aa3180), (*types.VirtualMachineVideoCard)(0xc8207ccfc0), (*types.VirtualMachineVMCIDevice)(0xc8201d7880), (*types.VirtualLsiLogicController)(0xc8207cd050), (*types.VirtualCdrom)(0xc821aa31d0), (*types.VirtualDisk)(0xc8204e1180), (*types.VirtualVmxnet3)(0xc8204e1220), (*types.VirtualDisk)(0xc8204e12c0), (*types.VirtualDisk)(0xc8204e1360)}}, CpuAllocation:(*types.ResourceAllocationInfo)(0xc8204d8480), MemoryAllocation:(*types.ResourceAllocationInfo)(0xc8204d84b0), LatencySensitivity:(*types.LatencySensitivity)(0xc82038e960), MemoryHotAddEnabled:(*bool)(0xc823b40876), CpuHotAddEnabled:(*bool)(0xc823b40885), CpuHotRemoveEnabled:(*bool)(0xc823b408b5), HotPlugMemoryLimit:2048, HotPlugMemoryIncrementSize:0, CpuAffinity:(*types.VirtualMachineAffinityInfo)(nil), MemoryAffinity:(*types.VirtualMachineAffinityInfo)(nil), NetworkShaper:(*types.VirtualMachineNetworkShaperInfo)(nil), ExtraConfig:[]types.BaseOptionValue{(*types.OptionValue)(0xc82038eee0), (*types.OptionValue)(0xc82038f140), (*types.OptionValue)(0xc82038f320), (*types.OptionValue)(0xc82038f4c0), (*types.OptionValue)(0xc82038f700), (*types.OptionValue)(0xc82038f880), (*types.OptionValue)(0xc82038fac0), (*types.OptionValue)(0xc82038fd20), (*types.OptionValue)(0xc82038ffa0), (*types.OptionValue)(0xc820492180), (*types.OptionValue)(0xc8204923c0), (*types.OptionValue)(0xc8204926c0), (*types.OptionValue)(0xc820492960), (*types.OptionValue)(0xc820492b60), (*types.OptionValue)(0xc820492e00), (*types.OptionValue)(0xc820493040), (*types.OptionValue)(0xc820493280), (*types.OptionValue)(0xc8204934c0), (*types.OptionValue)(0xc820493680), (*types.OptionValue)(0xc8204938a0), (*types.OptionValue)(0xc820493ac0), (*types.OptionValue)(0xc820493c80), (*types.OptionValue)(0xc820493f20), (*types.OptionValue)(0xc820218140), (*types.OptionValue)(0xc820218340), (*types.OptionValue)(0xc820218580), (*types.OptionValue)(0xc820218820), (*types.OptionValue)(0xc8202189e0), (*types.OptionValue)(0xc820218ba0), (*types.OptionValue)(0xc820218d60), (*types.OptionValue)(0xc820218f60), (*types.OptionValue)(0xc820219160), (*types.OptionValue)(0xc820219380), (*types.OptionValue)(0xc8202195c0), (*types.OptionValue)(0xc820219800), (*types.OptionValue)(0xc820219ac0), (*types.OptionValue)(0xc820219dc0), (*types.OptionValue)(0xc820219fc0), (*types.OptionValue)(0xc8201562a0), (*types.OptionValue)(0xc820156560), (*types.OptionValue)(0xc8201567c0), (*types.OptionValue)(0xc820156a00)}, CpuFeatureMask:[]types.HostCpuIdInfo(nil), DatastoreUrl:[]types.VirtualMachineConfigInfoDatastoreUrlPair{types.VirtualMachineConfigInfoDatastoreUrlPair{DynamicData:types.DynamicData{}, Name:"VMFS_TGT", Url:"/vmfs/volumes/568d2f37-f34ad8c6-d7e5-78e3b50fcde0"}}, SwapPlacement:"inherit", BootOptions:(*types.VirtualMachineBootOptions)(0xc821aa3360), FtInfo:types.BaseFaultToleranceConfigInfo(nil), RepConfig:(*types.ReplicationConfigSpec)(nil), VAppConfig:(*types.VmConfigInfo)(0xc82060e750), VAssertsEnabled:(*bool)(0xc823b41f69), ChangeTrackingEnabled:(*bool)(0xc823b41f9d), Firmware:"bios", MaxMksConnections:40, GuestAutoLockEnabled:(*bool)(0xc823b41fea), ManagedBy:(*types.ManagedByInfo)(nil), MemoryReservationLockedToMax:(*bool)(0xc823b41ffd), InitialOverhead:(*types.VirtualMachineConfigInfoOverheadInfo)(0xc823b3c170), NestedHVEnabled:(*bool)(0xc823b3c199), VPMCEnabled:(*bool)(0xc823b3c20b), ScheduledHardwareUpgradeInfo:(*types.ScheduledHardwareUpgradeInfo)(0xc8204d66c0), ForkConfigInfo:(*types.VirtualMachineForkConfigInfo)(nil), VFlashCacheReservation:0, VmxConfigChecksum:[]uint8(nil), MessageBusTunnelEnabled:(*bool)(nil), VmStorageObjectId:"", SwapStorageObjectId:""}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] mvm.Guest.Net - []types.GuestNicInfo{types.GuestNicInfo{DynamicData:types.DynamicData{}, Network:"", IpAddress:[]string{"172.17.0.1"}, MacAddress:"02:42:a8:e6:ac:b4", Connected:true, DeviceConfigId:-1, DnsConfig:(*types.NetDnsConfigInfo)(nil), IpConfig:(*types.NetIpConfigInfo)(0xc8204d84e0), NetBIOSConfig:types.BaseNetBIOSConfigInfo(nil)}, types.GuestNicInfo{DynamicData:types.DynamicData{}, Network:"NET_IF", IpAddress:[]string{"10.90.80.91", "fe80::250:56ff:fea9:129"}, MacAddress:"00:50:56:a9:01:29", Connected:true, DeviceConfigId:4000, DnsConfig:(*types.NetDnsConfigInfo)(nil), IpConfig:(*types.NetIpConfigInfo)(0xc8204d8510), NetBIOSConfig:types.BaseNetBIOSConfigInfo(nil)}}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] resourceVSphereVirtualMachineRead - Analyzing disk: [VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com.vmdk
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] mvm.Guest.Net - []types.GuestNicInfo{types.GuestNicInfo{DynamicData:types.DynamicData{}, Network:"", IpAddress:[]string{"172.17.0.1"}, MacAddress:"02:42:a8:e6:ac:b4", Connected:true, DeviceConfigId:-1, DnsConfig:(*types.NetDnsConfigInfo)(nil), IpConfig:(*types.NetIpConfigInfo)(0xc8204d84e0), NetBIOSConfig:types.BaseNetBIOSConfigInfo(nil)}, types.GuestNicInfo{DynamicData:types.DynamicData{}, Network:"NET_IF", IpAddress:[]string{"10.90.80.91", "fe80::250:56ff:fea9:129"}, MacAddress:"00:50:56:a9:01:29", Connected:true, DeviceConfigId:4000, DnsConfig:(*types.NetDnsConfigInfo)(nil), IpConfig:(*types.NetIpConfigInfo)(0xc8204d8510), NetBIOSConfig:types.BaseNetBIOSConfigInfo(nil)}}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] resourceVSphereVirtualMachineRead - Analyzing disk: [VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com.vmdk
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] disks: []map[string]interface {}{map[string]interface {}{"type":"eager_zeroed", "key":0, "bootable":false, "template":"templates/my-template-v123", "datastore":"VMFS_DIR/VMFS_TGT", "keep_on_remove":false, "uuid":"", "name":"", "iops":0, "vmdk":"", "controller_type":"scsi", "size":0}}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] resourceVSphereVirtualMachineRead - Analyzing disk: [VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com-vardisk.vmdk
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] disks: []map[string]interface {}{map[string]interface {}{"type":"eager_zeroed", "key":0, "bootable":false, "keep_on_remove":false, "uuid":"", "name":"", "iops":0, "vmdk":"", "controller_type":"scsi", "template":"templates/my-template-v123", "datastore":"VMFS_DIR/VMFS_TGT", "size":0}}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] resourceVSphereVirtualMachineRead - Analyzing disk: [VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com-dockerdisk.vmdk
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] disks: []map[string]interface {}{map[string]interface {}{"datastore":"VMFS_DIR/VMFS_TGT", "keep_on_remove":false, "uuid":"", "name":"", "iops":0, "vmdk":"", "controller_type":"scsi", "template":"templates/my-template-v123", "size":0, "type":"eager_zeroed", "key":0, "bootable":false}}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] v.Network - "NET_IF"
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] p.String - "10.90.80.91"
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] ip.PrefixLength - 25
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] networkInterface: map[string]interface {}{"mac_address":"00:50:56:a9:01:29", "ipv4_address":"10.90.80.91", "ipv4_prefix_length":25, "label":"NET_IF"}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] p.String - "fe80::250:56ff:fea9:129"
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] ip.PrefixLength - 64
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] networkInterface: map[string]interface {}{"ipv4_address":"10.90.80.91", "ipv4_prefix_length":25, "ipv6_address":"fe80::250:56ff:fea9:129", "ipv6_prefix_length":64, "label":"NET_IF", "mac_address":"00:50:56:a9:01:29"}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] networkInterface: map[string]interface {}{"ipv6_address":"fe80::250:56ff:fea9:129", "ipv6_prefix_length":64, "label":"NET_IF", "mac_address":"00:50:56:a9:01:29", "ipv4_address":"10.90.80.91", "ipv4_prefix_length":25}
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: vsphere-provider (internal) 2016/08/18 00:38:44 [DEBUG] ipv4_gateway of device id 1: 10.90.80.126
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: ^[[31mError applying plan:

1 error(s) occurred:

* vsphere_virtual_machine.generic_vmware: unexpected EOF

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.^[[0m^[[0m
panic: runtime error: index out of range
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: goroutine 114 [running]:
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: panic(0x2fb1ba0, 0xc8200101c0)
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:     /opt/go/src/runtime/panic.go:481 +0x3e6
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: github.com/hashicorp/terraform/builtin/providers/vsphere.resourceVSphereVirtualMachineRead(0xc8203f0900, 0x37041e0, 0xc8204cfe90, 0x0, 0x0)
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:     /opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/vsphere/resource_vsphere_virtual_machine.go:1049 +0x4368
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: github.com/hashicorp/terraform/builtin/providers/vsphere.resourceVSphereVirtualMachineCreate(0xc8203f0900, 0x37041e0, 0xc8204cfe90, 0x0, 0x0)
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:     /opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/vsphere/resource_vsphere_virtual_machine.go:895 +0x4523
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc8203227e0, 0xc8205473c0, 0xc8201abbc0, 0x37041e0, 0xc8204cfe90, 0xc82039bf01, 0x0, 0x0)
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:     /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:162 +0x29a
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc8202ce990, 0xc820547380, 0xc8205473c0, 0xc8201abbc0, 0x1, 0x0, 0x0)
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:     /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:176 +0x1ed
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Apply(0xc8204ae080, 0xc8201ab540, 0xc8204a4590, 0x0, 0x0)
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:     /opt/gopath/src/github.com/hashicorp/terraform/plugin/resource_provider.go:460 +0x76
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: reflect.Value.call(0x2c6c380, 0x36505b8, 0x13, 0x37c3bf8, 0x4, 0xc8229f9eb8, 0x3, 0x3, 0x0, 0x0, ...)
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:     /opt/go/src/reflect/value.go:435 +0x120d
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: reflect.Value.Call(0x2c6c380, 0x36505b8, 0x13, 0xc8229f9eb8, 0x3, 0x3, 0x0, 0x0, 0x0)
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:     /opt/go/src/reflect/value.go:303 +0xb1
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: net/rpc.(*service).call(0xc8204ac140, 0xc8204ac100, 0xc8204cf150, 0xc820094200, 0xc82046a900, 0x2540ae0, 0xc8201ab540, 0x16, 0x2540b40, 0xc8204a4590, ...)
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:     /opt/go/src/net/rpc/server.go:383 +0x1c2
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig: created by net/rpc.(*Server).ServeCodec
2016/08/18 00:38:44 [DEBUG] plugin: terraform.orig:     /opt/go/src/net/rpc/server.go:477 +0x49d
2016/08/18 00:38:44 [DEBUG] root.generic-provisioner: eval: *terraform.EvalWriteState
2016/08/18 00:38:44 [DEBUG] root.generic-provisioner: eval: *terraform.EvalApplyProvisioners
2016/08/18 00:38:44 [DEBUG] root.generic-provisioner: eval: *terraform.EvalIf
2016/08/18 00:38:44 [DEBUG] root.generic-provisioner: eval: *terraform.EvalWriteState
2016/08/18 00:38:44 [DEBUG] root.generic-provisioner: eval: *terraform.EvalWriteDiff
2016/08/18 00:38:44 [DEBUG] root.generic-provisioner: eval: *terraform.EvalApplyPost
2016/08/18 00:38:44 [ERROR] root.generic-provisioner: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* vsphere_virtual_machine.generic_vmware: unexpected EOF
2016/08/18 00:38:44 [ERROR] root.generic-provisioner: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* vsphere_virtual_machine.generic_vmware: unexpected EOF
2016/08/18 00:38:44 [ERROR] root.generic-provisioner: eval: *terraform.EvalOpFilter, err: 1 error(s) occurred:

* vsphere_virtual_machine.generic_vmware: unexpected EOF
2016/08/18 00:38:44 [ERROR] root.generic-provisioner: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* vsphere_virtual_machine.generic_vmware: unexpected EOF
2016/08/18 00:38:44 [TRACE] [walkApply] Exiting eval tree: vsphere_virtual_machine.generic_vmware
```

Post-patch:

```
2016/08/18 00:55:11 [INFO] Created virtual machine: templates/my-new-vm.dc.domain.com
2016/08/18 00:55:11 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:55:11 [DEBUG] virtual machine resource data: &schema.ResourceData{schema:map[string]*schema.Schema{"datacenter":(*schema.Schema)(0xc820402ee0), "gateway":(*schema.Schema)(0xc820403260), "disk":(*schema.Schema)(0xc820403c00), "network_interface":(*schema.Schema)(0xc820403b20), "name":(*schema.Schema)(0xc820402a80), "memory":(*schema.Schema)(0xc820402d20), "cluster":(*schema.Schema)(0xc820402fc0), "custom_configuration_parameters":(*schema.Schema)(0xc820403960), "linked_clone":(*schema.Schema)(0xc820403180), "domain":(*schema.Schema)(0xc820403340), "time_zone":(*schema.Schema)(0xc820403420), "dns_suffixes":(*schema.Schema)(0xc820403500), "folder":(*schema.Schema)(0xc820402b60), "vcpu":(*schema.Schema)(0xc820402c40), "memory_reservation":(*schema.Schema)(0xc820402e00), "resource_pool":(*schema.Schema)(0xc8204030a0), "dns_servers":(*schema.Schema)(0xc8204035e0), "enable_disk_uuid":(*schema.Schema)(0xc8204037a0), "uuid":(*schema.Schema)(0xc820403880), "skip_customization":(*schema.Schema)(0xc8204036c0), "windows_opt_config":(*schema.Schema)(0xc820403a40), "cdrom":(*schema.Schema)(0xc820403ce0)}, config:(*terraform.ResourceConfig)(nil), state:(*terraform.InstanceState)(nil), diff:(*terraform.InstanceDiff)(0xc82034d780), meta:map[string]string(nil), multiReader:(*schema.MultiLevelFieldReader)(0xc82034df20), setWriter:(*schema.MapFieldWriter)(0xc82034de80), newState:(*terraform.InstanceState)(0xc820245340), partial:false, partialMap:map[string]struct {}(nil), once:sync.Once{m:sync.Mutex{state:0, sema:0x0}, done:0x1}, isNew:true}
...
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] Successfully waited for interfaces to appear
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] Datacenter - &object.Datacenter{Common:object.Common{c:(*vim25.Client)(0xc8201f8780), r:types.ManagedObjectReference{Type:"Datacenter", Value:"datacenter-2"}}}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] mvm.Summary.Config - types.VirtualMachineConfigSummary{DynamicData:types.DynamicData{}, Name:"my-new-vm.dc.domain.com", Template:false, VmPathName:"[VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com.vmx", MemorySizeMB:2048, CpuReservation:0, MemoryReservation:0, NumCpu:2, NumEthernetCards:1, NumVirtualDisks:3, Uuid:"42297081-9aca-e643-c0f7-e806f21256c2", InstanceUuid:"50298333-2904-6209-8042-b04cb195423b", GuestId:"centos64Guest", GuestFullName:"CentOS 4/5/6/7 (64-bit)", Annotation:"", Product:(*types.VAppProductInfo)(0xc8239c4960), InstallBootRequired:(*bool)(0xc8208ffc8d), FtInfo:types.BaseFaultToleranceConfigInfo(nil), ManagedBy:(*types.ManagedByInfo)(nil)}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] Datacenter - &object.Datacenter{Common:object.Common{c:(*vim25.Client)(0xc8201f8780), r:types.ManagedObjectReference{Type:"Datacenter", Value:"datacenter-2"}}}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] mvm.Summary.Config - types.VirtualMachineConfigSummary{DynamicData:types.DynamicData{}, Name:"my-new-vm.dc.domain.com", Template:false, VmPathName:"[VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com.vmx", MemorySizeMB:2048, CpuReservation:0, MemoryReservation:0, NumCpu:2, NumEthernetCards:1, NumVirtualDisks:3, Uuid:"42297081-9aca-e643-c0f7-e806f21256c2", InstanceUuid:"50298333-2904-6209-8042-b04cb195423b", GuestId:"centos64Guest", GuestFullName:"CentOS 4/5/6/7 (64-bit)", Annotation:"", Product:(*types.VAppProductInfo)(0xc8239c4960), InstallBootRequired:(*bool)(0xc8208ffc8d), FtInfo:types.BaseFaultToleranceConfigInfo(nil), ManagedBy:(*types.ManagedByInfo)(nil)}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] mvm.Summary.Config - &types.VirtualMachineConfigInfo{DynamicData:types.DynamicData{}, ChangeVersion:"2016-08-17T23:55:34.752323Z", Modified:time.Time{sec:62135596800, nsec:0, loc:(*time.Location)(0x5871820)}, Name:"my-new-vm.dc.domain.com", GuestFullName:"CentOS 4/5/6/7 (64-bit)", Version:"vmx-10", Uuid:"42297081-9aca-e643-c0f7-e806f21256c2", InstanceUuid:"50298333-2904-6209-8042-b04cb195423b", NpivNodeWorldWideName:[]int64(nil), NpivPortWorldWideName:[]int64(nil), NpivWorldWideNameType:"", NpivDesiredNodeWwns:0, NpivDesiredPortWwns:0, NpivTemporaryDisabled:(*bool)(0xc820b389a8), NpivOnNonRdmDisks:(*bool)(nil), LocationId:"564d72d4-da61-1c38-a438-42892b604202", Template:false, GuestId:"centos64Guest", AlternateGuestName:"", Annotation:"", Files:types.VirtualMachineFileInfo{DynamicData:types.DynamicData{}, VmPathName:"[VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com.vmx", SnapshotDirectory:"[VMFS_TGT] my-new-vm.dc.domain.com_1/", SuspendDirectory:"[VMFS_TGT] my-new-vm.dc.domain.com_1/", LogDirectory:"[VMFS_TGT] my-new-vm.dc.domain.com_1/", FtMetadataDirectory:""}, Tools:(*types.ToolsConfigInfo)(0xc820b35980), Flags:types.VirtualMachineFlagInfo{DynamicData:types.DynamicData{}, DisableAcceleration:(*bool)(0xc820b39365), EnableLogging:(*bool)(0xc820b393e5), UseToe:(*bool)(0xc820b3944a), RunWithDebugInfo:(*bool)(0xc820b394a5), MonitorType:"release", HtSharing:"any", SnapshotDisabled:(*bool)(0xc820b396ab), SnapshotLocked:(*bool)(0xc820b3974d), DiskUuidEnabled:(*bool)(0xc820b397ed), VirtualMmuUsage:"automatic", VirtualExecUsage:"hvAuto", SnapshotPowerOffBehavior:"powerOff", RecordReplayEnabled:(*bool)(0xc820b39a20), FaultToleranceType:""}, ConsolePreferences:(*types.VirtualMachineConsolePreferences)(nil), DefaultPowerOps:types.VirtualMachineDefaultPowerOpInfo{DynamicData:types.DynamicData{}, PowerOffType:"soft", SuspendType:"soft", ResetType:"soft", DefaultPowerOffType:"soft", DefaultSuspendType:"hard", DefaultResetType:"soft", StandbyAction:"checkpoint"}, Hardware:types.VirtualHardware{DynamicData:types.DynamicData{}, NumCPU:2, NumCoresPerSocket:1, MemoryMB:2048, VirtualICH7MPresent:(*bool)(0xc820b39f2c), VirtualSMCPresent:(*bool)(0xc820b39f9d), Device:[]types.BaseVirtualDevice{(*types.VirtualIDEController)(0xc820204770), (*types.VirtualIDEController)(0xc8202047e0), (*types.VirtualPS2Controller)(0xc820204850), (*types.VirtualPCIController)(0xc8202048c0), (*types.VirtualSIOController)(0xc820204930), (*types.VirtualKeyboard)(0xc821e43d10), (*types.VirtualPointingDevice)(0xc8207e43c0), (*types.VirtualMachineVideoCard)(0xc8200af290), (*types.VirtualMachineVMCIDevice)(0xc8202049a0), (*types.VirtualLsiLogicController)(0xc8200af320), (*types.VirtualCdrom)(0xc8207e4640), (*types.VirtualDisk)(0xc8239c46e0), (*types.VirtualVmxnet3)(0xc8239c4780), (*types.VirtualDisk)(0xc8239c4820), (*types.VirtualDisk)(0xc8239c48c0)}}, CpuAllocation:(*types.ResourceAllocationInfo)(0xc820a769f0), MemoryAllocation:(*types.ResourceAllocationInfo)(0xc820a76a20), LatencySensitivity:(*types.LatencySensitivity)(0xc8218055c0), MemoryHotAddEnabled:(*bool)(0xc82096db96), CpuHotAddEnabled:(*bool)(0xc82096dbf5), CpuHotRemoveEnabled:(*bool)(0xc82096dc65), HotPlugMemoryLimit:2048, HotPlugMemoryIncrementSize:0, CpuAffinity:(*types.VirtualMachineAffinityInfo)(nil), MemoryAffinity:(*types.VirtualMachineAffinityInfo)(nil), NetworkShaper:(*types.VirtualMachineNetworkShaperInfo)(nil), ExtraConfig:[]types.BaseOptionValue{(*types.OptionValue)(0xc8218059a0), (*types.OptionValue)(0xc821805b40), (*types.OptionValue)(0xc821805ce0), (*types.OptionValue)(0xc821805e40), (*types.OptionValue)(0xc821805fc0), (*types.OptionValue)(0xc8208900e0), (*types.OptionValue)(0xc820890260), (*types.OptionValue)(0xc820890400), (*types.OptionValue)(0xc820890580), (*types.OptionValue)(0xc820890700), (*types.OptionValue)(0xc820890880), (*types.OptionValue)(0xc8208909e0), (*types.OptionValue)(0xc820890b80), (*types.OptionValue)(0xc820890d00), (*types.OptionValue)(0xc820890ec0), (*types.OptionValue)(0xc8208910a0), (*types.OptionValue)(0xc820891280), (*types.OptionValue)(0xc820891480), (*types.OptionValue)(0xc8208915c0), (*types.OptionValue)(0xc820891780), (*types.OptionValue)(0xc8208918e0), (*types.OptionValue)(0xc820891aa0), (*types.OptionValue)(0xc820891c20), (*types.OptionValue)(0xc820891de0), (*types.OptionValue)(0xc820891f80), (*types.OptionValue)(0xc8218160e0), (*types.OptionValue)(0xc821816240), (*types.OptionValue)(0xc821816360), (*types.OptionValue)(0xc8218164c0), (*types.OptionValue)(0xc8218165e0), (*types.OptionValue)(0xc821816740), (*types.OptionValue)(0xc8218168a0), (*types.OptionValue)(0xc821816a20), (*types.OptionValue)(0xc821816b80), (*types.OptionValue)(0xc821816ce0), (*types.OptionValue)(0xc821816e40), (*types.OptionValue)(0xc821816fa0), (*types.OptionValue)(0xc8218170a0), (*types.OptionValue)(0xc821817200), (*types.OptionValue)(0xc821817320), (*types.OptionValue)(0xc821817440), (*types.OptionValue)(0xc821817560)}, CpuFeatureMask:[]types.HostCpuIdInfo(nil), DatastoreUrl:[]types.VirtualMachineConfigInfoDatastoreUrlPair{types.VirtualMachineConfigInfoDatastoreUrlPair{DynamicData:types.DynamicData{}, Name:"VMFS_TGT", Url:"/vmfs/volumes/568d2f37-f34ad8c6-d7e5-78e3b50fcde0"}}, SwapPlacement:"inherit", BootOptions:(*types.VirtualMachineBootOptions)(0xc8207e4f00), FtInfo:types.BaseFaultToleranceConfigInfo(nil), RepConfig:(*types.ReplicationConfigSpec)(nil), VAppConfig:(*types.VmConfigInfo)(0xc820090750), VAssertsEnabled:(*bool)(0xc82093dee9), ChangeTrackingEnabled:(*bool)(0xc82093df8d), Firmware:"bios", MaxMksConnections:40, GuestAutoLockEnabled:(*bool)(0xc82093806a), ManagedBy:(*types.ManagedByInfo)(nil), MemoryReservationLockedToMax:(*bool)(0xc8209380cd), InitialOverhead:(*types.VirtualMachineConfigInfoOverheadInfo)(0xc8209381b0), NestedHVEnabled:(*bool)(0xc820938219), VPMCEnabled:(*bool)(0xc82093835b), ScheduledHardwareUpgradeInfo:(*types.ScheduledHardwareUpgradeInfo)(0xc822488ec0), ForkConfigInfo:(*types.VirtualMachineForkConfigInfo)(nil), VFlashCacheReservation:0, VmxConfigChecksum:[]uint8(nil), MessageBusTunnelEnabled:(*bool)(nil), VmStorageObjectId:"", SwapStorageObjectId:""}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] mvm.Guest.Net - []types.GuestNicInfo{types.GuestNicInfo{DynamicData:types.DynamicData{}, Network:"", IpAddress:[]string{"172.17.0.1"}, MacAddress:"02:42:1d:24:0e:23", Connected:true, DeviceConfigId:-1, DnsConfig:(*types.NetDnsConfigInfo)(nil), IpConfig:(*types.NetIpConfigInfo)(0xc820a76a80), NetBIOSConfig:types.BaseNetBIOSConfigInfo(nil)}, types.GuestNicInfo{DynamicData:types.DynamicData{}, Network:"NET_IF", IpAddress:[]string{"10.90.80.91", "fe80::250:56ff:fea9:281f"}, MacAddress:"00:50:56:a9:28:1f", Connected:true, DeviceConfigId:4000, DnsConfig:(*types.NetDnsConfigInfo)(nil), IpConfig:(*types.NetIpConfigInfo)(0xc820a76ae0), NetBIOSConfig:types.BaseNetBIOSConfigInfo(nil)}}2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] resourceVSphereVirtualMachineRead - Analyzing disk: [VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com-vardisk.vmdk
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] disks: []map[string]interface {}{map[string]interface {}{"keep_on_remove":false, "uuid":"", "template":"templates/my-template-v123", "type":"eager_zeroed", "key":0, "size":0, "bootable":false, "controller_type":"scsi", "name":"", "iops":0, "datastore":"VMFS_DIR/VMFS_TGT", "vmdk":""}}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] resourceVSphereVirtualMachineRead - Analyzing disk: [VMFS_TGT] my-new-vm.dc.domain.com_1/my-new-vm.dc.domain.com-dockerdisk.vmdk
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] disks: []map[string]interface {}{map[string]interface {}{"uuid":"", "template":"templates/my-template-v123", "type":"eager_zeroed", "key":0, "size":0, "keep_on_remove":false, "controller_type":"scsi", "name":"", "iops":0, "datastore":"VMFS_DIR/VMFS_TGT", "vmdk":"", "bootable":false}}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] v.Network - "NET_IF"
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] p.String - "10.90.80.91"
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] ip.PrefixLength - 25
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] networkInterface: map[string]interface {}{"ipv4_address":"10.90.80.91", "ipv4_prefix_length":25, "label":"NET_IF", "mac_address":"00:50:56:a9:28:1f"}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] p.String - "fe80::250:56ff:fea9:281f"
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] ip.PrefixLength - 64
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] networkInterface: map[string]interface {}{"ipv4_prefix_length":25, "ipv6_address":"fe80::250:56ff:fea9:281f", "ipv6_prefix_length":64, "label":"NET_IF", "mac_address":"00:50:56:a9:28:1f", "ipv4_address":"10.90.80.91"}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] networkInterface: map[string]interface {}{"label":"NET_IF", "mac_address":"00:50:56:a9:28:1f", "ipv4_address":"10.90.80.91", "ipv4_prefix_length":25, "ipv6_address":"fe80::250:56ff:fea9:281f", "ipv6_prefix_length":64}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] processing ipv4_gateway of device id 1 ("1") for types.NetIpRouteConfigInfoGateway{DynamicData:types.DynamicData{}, IpAddress:"10.90.80.126", Device:"1"}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] ipv4_gateway of device id 0: 10.90.80.126
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] gateway IP: 10.90.80.126
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] networkInterfaces: []map[string]interface {}{map[string]interface {}{"ipv6_address":"fe80::250:56ff:fea9:281f", "ipv6_prefix_length":64, "label":"NET_IF", "mac_address":"00:50:56:a9:28:1f", "ipv4_address":"10.90.80.91", "ipv4_prefix_length":25}}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] device: map[string]interface {}{"label":"NET_IF", "mac_address":"00:50:56:a9:28:1f", "ipv4_address":"10.90.80.91", "ipv4_prefix_length":25, "ipv6_address":"fe80::250:56ff:fea9:281f", "ipv6_prefix_length":64}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] networkInterfaces: []map[string]interface {}{map[string]interface {}{"label":"NET_IF", "mac_address":"00:50:56:a9:28:1f", "ipv4_address":"10.90.80.91", "ipv4_prefix_length":25, "ipv6_address":"fe80::250:56ff:fea9:281f", "ipv6_prefix_length":64, "ipv4_gateway":"10.90.80.126"}}
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] ip address: 10.90.80.91
2016/08/18 00:56:27 [DEBUG] plugin: terraform: vsphere-provider (internal) 2016/08/18 00:56:27 [DEBUG] "VMFS_TGT"
2016/08/18 00:56:27 [DEBUG] root.generic-provisioner: eval: *terraform.EvalWriteState
2016/08/18 00:56:27 [DEBUG] root.generic-provisioner: eval: *terraform.EvalApplyProvisioners
2016/08/18 00:56:27 [DEBUG] root.generic-provisioner: eval: *terraform.EvalIf
2016/08/18 00:56:27 [DEBUG] root.generic-provisioner: eval: *terraform.EvalWriteState
2016/08/18 00:56:27 [DEBUG] root.generic-provisioner: eval: *terraform.EvalWriteDiff
2016/08/18 00:56:27 [DEBUG] root.generic-provisioner: eval: *terraform.EvalApplyPost
2016/08/18 00:56:27 [DEBUG] root.generic-provisioner: eval: *terraform.EvalUpdateStateHook
^[[0m^[[1mmodule.generic-provisioner.vsphere_virtual_machine.generic_vmware: Creation complete^[[21m^[[0m
```

Maybe you could try the change in the PR and confirm whether it fixes your issue?

--- a/builtin/providers/vsphere/resource_vsphere_virtual_machine.go
+++ b/builtin/providers/vsphere/resource_vsphere_virtual_machine.go
@@ -1053,6 +1053,9 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 						}
 						if gatewaySetting != "" {
 							deviceID, err := strconv.Atoi(route.Gateway.Device)
+							if len(networkInterfaces) == 1 {
+								deviceID = 0
+							}
 							if err != nil {
 								log.Printf("[WARN] error at processing %s of device id %#v: %#v", gatewaySetting, route.Gateway.Device, err)
 							} else {

